### PR TITLE
WindowsPort.md: Switched to clang-cl

### PR DESCRIPTION
**Pull request by fujii**
Updated the document for the clang-cl migration.

### CODE REVIEW

**Review by donny-dont — COMMENTED** [review]
I'm not against clang-cl being the preferred method to compile WebKit on Windows. Its my understanding that there are some intrinsics in clang that improve the performance of JSC on Windows. That's good stuff for anyone shipping the Windows port.

I don't think we should get rid of it fully to make it easier for people to start contributing. Both `clang-cl` and `cl` have had times when they fail to compile things on Windows so I don't see one as more stable than the other. I also don't think a monoculture of compilers is a good thing.

File: docs/Ports/WindowsPort.md [hunk 1]
```diff
@@ -7,27 +7,23 @@ It supports only 64 bit Windows.
 
 ## Installing Development Tools
 
-You need CMake, Perl, Python, Ruby, gperf, the latest Windows SDK, and Visual Studio 2022 to build Windows port.
-You can use [Chocolatey](https://chocolatey.org/) to install the tools.
+It's recommended to [install the latest Visual Studio and "Desktop development with C++" workload](https://learn.microsoft.com/en-us/cpp/build/vscpp-step-0-installation) for the nice debugger.
```
[the remark this answers is not in the document]
Comment (fujii) [reply docs/Ports/WindowsPort.md:10]: Windows SDK includes a simple debugger `WinDbg`. I mean VS is better than it.

File: docs/Ports/WindowsPort.md [hunk 2]
```diff
@@ -113,10 +112,13 @@ You can run programs under a debugger with [this instruction](../Build & Debug/D
...
 ### Building from within Visual Studio
 
+You can use CMake Visual Studio generator instead of Ninja generator.
+Install [the LLVM extension](https://learn.microsoft.com/en-us/cpp/build/clang-support-msbuild) of MSBuild.
+
 In the WinKit command prompt,
 
 ```
-perl Tools/Scripts/build-webkit --release --no-ninja --generate-project-only
+perl Tools/Scripts/build-webkit --release --no-ninja --cmakeargs="-T ClangCL" --generate-project-only
```
[the remark this answers is not in the document]
Comment (fujii) [reply docs/Ports/WindowsPort.md:121]: Agreed. Will fix it in https://bugs.webkit.org/show_bug.cgi?id=262506

File: docs/Ports/WindowsPort.md [hunk 3]
```diff
@@ -213,24 +215,3 @@ OK
...
-
-Open Visual Studio Command Prompt, and invoke the following commands.
-
-```
-set CC=clang-cl
-set CXX=clang-cl
-perl Tools\Scripts\build-webkit --release --ninja
-```
-
-clang-cl builds are experimental, see [Bug 171618](https://bugs.webkit.org/show_bug.cgi?id=171618) for the current status.
```
[the remark this answers is not in the document]
Comment (fujii) [reply docs/Ports/WindowsPort.md:236]: I think "Building from within Visual Studio" section is the one you want.

**Comment by fujii** [comment]
> I don't think we should get rid of it fully to make it easier for people to start contributing.

Chromium and Firefox have already dropped MSVC support. I think WebKit should drop it to improve productivity not only for us but also for all WebKit team.

> Both clang-cl and cl have had times when they fail to compile things on Windows so I don't see one as more stable than the other.

MSVC also had the bad times. If new clang-cl releases will have a bug, we should be stick to older clang-cl, report the bug.

> I also don't think a monoculture of compilers is a good thing.

We can restore the MSVC support if it improves interoperability and supports GCC extensions.

File: docs/Ports/WindowsPort.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (fujii) [reply docs/Ports/WindowsPort.md]: Removed the `--cmakeargs` switch.

File: docs/Ports/WindowsPort.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (donny-dont) [reply docs/Ports/WindowsPort.md:10]: I would drop the nice debugger part and just make it this

> Install the latest Visual Studio with the Desktop Development with C++ workload.

[the remark this answers is not in the document]
Comment (fujii) [reply docs/Ports/WindowsPort.md]: Fixed.

**Comment by fujii** [comment]
A Firefox developer was also thinking about the compiler monoculture.
https://blog.mozilla.org/nfroyd/2018/05/29/when-implementation-monoculture-right-thing/

**Review by donny-dont — APPROVED** [review]

**Pull request merged**